test/functionalTest/fixContentLengths.py

```diff
@@ -0,0 +1,299 @@
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+
+from getopt import getopt, GetoptError
+
+import os
+import sys
+import re
+import tempfile
+from operator import add
+
+__author__ = 'fermin'
+
+
+#####################
+# Functions program #
+#####################
+
+def msg(m):
+    """
+    Print message if verbose mode is enabled
+
+    :param m: message to print
+    """
+    global verbose
+
+    if verbose:
+        print m
+
+
+def usage_and_exit(m):
+    """
+    Print usage message and exit"
+
+    :param m: optional error message to print
+    """
+
+    if m != '':
+        print m
+        print
+
+    usage()
+    sys.exit(1)
+
+
+def usage():
+    """
+    Print usage message
+    """
+
+    print 'Modifies each "Content-Length" line in .test files, based on the same values in the corresponding .out file'
+    print ''
+    print 'Usage: %s -f <file> [-d] [-v] [-u]' % os.path.basename(__file__)
+    print ''
+    print 'Parameters:'
+    print '  -f <file>: .test or .out file to modify. Alternatively, it can be a directory in which case'
+    print '             all the .test/.out files in that directory are checked, recursively.'
+    print '  -d: dry-run mode, i.e. .test files are not modified'
+    print '  -v: verbose mode'
+    print '  -u: print this usage message'
+
+
+def patch_content_lengths(file_name, cl):
+    """
+    Patch the Content-Length lines in file_name with the data comming from the cl argument,
+    assuming it is a list of pairs as returned by content_length_extract(), i.e. using the
+    second argument in each pair.
+
+    The patch is based in the technique of writing a temporal file, then replace the original
+    one with it.
+
+    It is assumed that cl is ok before invoking this function, i.e. check_same_lines() has been
+    previously named.
+
+    :param file_name: the .test file to patch
+    :param cl: the list of pair to be used for patching
+    """
+
+    msg('  - patching file %s' % file_name)
+
+    n = 0
+    file_temp = tempfile.NamedTemporaryFile(delete=False)
+    for line in open(file_name):
+        m = re.match('^Content-Length: \d+', line)
+        if m is not None:
+            file_temp.write('Content-Length: %d\n' % cl[n][1])
+            msg ('    - patching "Content-Length: %d"' % cl[n][1])
+            n += 1
+        else:
+            file_temp.write(line)
+
+    file_temp.close()
+
+    # Remove old file, replacing by the edited one
+    os.remove(file_name)
+    os.rename(file_temp.name, file_name)
+
+
+def content_length_extract(file_name):
+    """
+    Open file passed as argument, looking for Content-Lengh lines. A list of pairs is removed, the
+    first item in each pair being a line number and the second item the Content-Lenght value
+
+    :param file_name: file to be processed
+    :return: a list of pairs as described above
+    """
+
+    r = []
+    l = 0
+    for line in open(file_name):
```

Comment:
Fixed (for this an other `open()` ocurrence) in 5765b576b7b694e8b0c4b5f65a8dfb669f7a17c4